File: spring-boot-kafka-demo/kafka-json-demo1/src/main/java/com/github/jccode/springbootkafkademo/jsondemo1/event/EventProducer.java
package com.github.jccode.springbootkafkademo.jsondemo1.event;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Component
public class EventProducer {

    @Autowired
    private KafkaTemplate<String, Event> template;

    public void produceEvents1() {
        for (int i = 0; i < 10; i++) {
            List<String> tokens = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                tokens.add(String.valueOf(j));
            }
            Event tokenEvent = new Event("tokens", tokens);
            template.send(Constant.EVENT_TOPIC, String.valueOf(i), tokenEvent);
        }
    }

    public void produceEvents2() {
        for (int i = 0; i < 10; i++) {
            Map<String, String> passwords = new HashMap<>();
            passwords.put("user_"+i, "pass"+(char)(97+i));
            Event passEvent = new Event("user_passwords", passwords);
            template.send(Constant.EVENT_TOPIC, String.valueOf(i), passEvent);
        }
    }
}

File: spring-boot-kafka-demo/kafka-json-demo1/src/main/scala/com/github/jccode/springbootkafkademo/jsondemo1/event/EventProducer.scala
package com.github.jccode.springbootkafkademo.jsondemo1.event

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

import scala.collection.JavaConverters._
import scala.collection.mutable.ListBuffer

@Component
class EventProducer(@Autowired private val template: KafkaTemplate[String, Event]) {

  def produceEvents1(): Unit = {
    for (i <- 0 to 10) {
      var tokens = ListBuffer[String]()
      for (j <- 0 to i) {
        tokens += s"$j"
      }
      // 传入kafka的时候,需要转成java的集合类. scala的集合序列化,反序列化有点问题.
      template.send(Constant.EventTopic, i+"", Event("tokens", tokens.asJava))
    }
  }

  def produceEvents2(): Unit = {
    for (i <- 0 to 10) {
      val passwords = Map[String, String](s"user_$i" -> s"pass${(97+i).toChar}")
      template.send(Constant.EventTopic, i.toString, Event("user_passwords", passwords.asJava))
    }
  }
}
